cpp/ql/src/semmle/code/cpp/Macro.qll

```diff
@@ -11,7 +11,7 @@ class Macro extends PreprocessorDirective, @ppd_define {
    */
   override string getHead() { preproctext(underlyingElement(this),result,_) }
 
-  override string getCanonicalQLClass() { result = "Macro" }
+  override string getCanonicalQLClass() { result = "Macro(" + this.getName() + ")" }
```

Comment:
Why add the macro name to the string? I don't see the value, but it might hurt performance, and it seems inconsistent with all the other `getCanonicalQLClass` overrides. If someone is looking for a string that identifies `this`, they already have `toString`.